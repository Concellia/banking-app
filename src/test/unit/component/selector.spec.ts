import { mount } from '@vue/test-utils'
import Selector from '../../../components/selector.vue'
import { test, describe, expect } from 'vitest'

const wrapper = mount(Selector, {
  propsData: {
    title: 'Test',
    isSelected: false,
    description: 'Describe'
  }
})

describe('Selector', () => {
  test('should render correctly', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })

  test('should emiton button click when the button is clicked', () => {
    const button = wrapper.find('.selector-radio-button')
    const clickEvent = new Event('click')
    button.element.dispatchEvent(clickEvent)
    expect(wrapper.emitted('on-item-select')).toBeTruthy()
  })
})
