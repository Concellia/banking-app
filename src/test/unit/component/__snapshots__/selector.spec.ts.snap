// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Selector > should render correctly 1`] = `"TestDescribe"`;

exports[`TextCard > should render correctly 1`] = `""`;

exports[`should render correctly 1`] = `"TestDescribe"`;
